### PR TITLE
[SPARK-37735][K8S] Add appId interface to KubernetesConf

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
Add `appId` interface to KubernetesConf

### Why are the changes needed?
The `appId` now can be only accessed in `KuberntesDriverConf` and `KubernetesExecutorConf`, but can't be accesssed in `KubernetesConf`.

Some user featurestep are using `KubernetesConf` as init constructor parameter in order to share the featurestep between driver and executor. One of cases is customized feature step (such as volcano, yunikorn) is using appId as job identification.

So, we'd better add appId to KubernetesConf to help such featurestep access appId.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT

### CODE REVIEW

**Comment by Yikun** [comment]
cc @dongjoon-hyun @holdenk

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala [hunk 1]
```diff
@@ -218,6 +218,14 @@ class KubernetesConfSuite extends SparkFunSuite {
     assert(driverConf.schedulerName === "driverScheduler")
   }
 
+  test("SPARK-36059: access appId in KubernetesConf") {
```
Comment (dongjoon-hyun) [inline resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala:221]: Given that the PR title is `SPARK-37735`, this `SPARK-36059` looks wrong to me.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
I left one minor comment on the test.

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala (hunk 1 above)
Comment (Yikun) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala]: @dongjoon-hyun Thanks, it was bad copy paste. Fixed.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs)
Thank you for updates.

**Comment by dongjoon-hyun** [comment]
cc @williamhyun , too

**Comment by Yikun** [comment]
```
starting black test...
black checks failed:
would reformat python/pyspark/shuffle.py
Oh no! 💥 💔 💥
1 file would be reformatted, 358 files would be left unchanged.
Please run 'dev/reformat-python' script.
```

Looks like need rebase

File: resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala [hunk 2]
```diff
@@ -218,6 +218,14 @@ class KubernetesConfSuite extends SparkFunSuite {
     assert(driverConf.schedulerName === "driverScheduler")
   }
 
+  test("SPARK-37735: access appId in KubernetesConf") {
+    val sparkConf = new SparkConf(false)
+    val execConf = KubernetesTestConf.createExecutorConf(sparkConf)
+    val driverConf = KubernetesTestConf.createExecutorConf(sparkConf)
```
Comment (williamhyun) [inline resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala:224]: Is `createExecutorConf` a typo?

Comment (dongjoon-hyun) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala:224]: Oh, nice catch. I missed this. Thanks, @williamhyun .

To @Yikun , please fix this by using `createDriverConf` here.

Comment (dongjoon-hyun) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala:224]: And, if possible, please put `driverConf` before `execConf`.

Comment (Yikun) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala:224]: Ah, thanks, will fix soon.

Comment (Yikun) [reply resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/KubernetesConfSuite.scala]: done

**Review by dongjoon-hyun — APPROVED** [review]

**Review by williamhyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master. Thank you, @Yikun and @williamhyun .

**Pull request closed without merging**